modules/light-clients/08-wasm/types/misbehaviour_handle_test.go

```diff
@@ -469,372 +481,128 @@ package types_test
+// 				clientMessage,
+// 			)
+
+// 			suite.Require().Equal(tc.foundMisbehaviour, foundMisbehaviour)
+// 		})
+// 	}
 // }
 
-/* func (suite *TypesTestSuite) TestCheckForMisbehaviourGrandpa() {
+func (suite *TypesTestSuite) TestCheckForMisbehaviour() {
 	var (
-		ok          bool
-		clientMsg   exported.ClientMessage
-		clientState exported.ClientState
+		foundMisbehaviour bool
+		panicErr          error
 	)
 
 	testCases := []struct {
-		name     string
-		malleate func()
-		expPass  bool
+		name                 string
+		malleate             func()
+		expFoundMisbehaviour bool
+		expPanic             bool
 	}{
 		{
-			"valid update no misbehaviour",
+			"no misbehaviour",
 			func() {
-				data, err := base64.StdEncoding.DecodeString(suite.testData["header"])
-				suite.Require().NoError(err)
-				clientMsg = &types.ClientMessage{
-					Data: data,
-				}
-
-				err = clientState.VerifyClientMessage(suite.ctx, suite.chainA.Codec, suite.store, clientMsg)
-				suite.Require().NoError(err)
+				suite.mockVM.RegisterQueryCallback(types.CheckForMisbehaviourMsg{}, func(codeID wasmvm.Checksum, env wasmvmtypes.Env, queryMsg []byte, store wasmvm.KVStore, goapi wasmvm.GoAPI, querier wasmvm.Querier, gasMeter wasmvm.GasMeter, gasLimit uint64, deserCost wasmvmtypes.UFraction) ([]byte, uint64, error) {
+					resp, err := json.Marshal(types.CheckForMisbehaviourResult{FoundMisbehaviour: false})
+					suite.Assert().NoError(err)
+					return resp, types.DefaultGasUsed, nil
+				})
 			},
 			false,
+			false,
 		},
 		{
-			"valid fork misbehaviour returns true",
-			func() {
-				data, err := base64.StdEncoding.DecodeString(suite.testData["header"])
-				suite.Require().NoError(err)
-				clientMsg = &types.ClientMessage{
-					Data: data,
-				}
-				// VerifyClientMessage must be run first
-				err = clientState.VerifyClientMessage(suite.ctx, suite.chainA.Codec, suite.store, clientMsg)
-				suite.Require().NoError(err)
-				clientState.UpdateState(suite.ctx, suite.chainA.Codec, suite.store, clientMsg)
-
-				// Reset client state to the previous for the test
-				suite.chainA.App.GetIBCKeeper().ClientKeeper.SetClientState(suite.ctx, grandpaClientID, clientState)
-
-				data, err = base64.StdEncoding.DecodeString(suite.testData["misbehaviour"])
-				suite.Require().NoError(err)
-				clientMsg = &types.ClientMessage{
-					Data: data,
-				}
-
-				err = clientState.VerifyClientMessage(suite.ctx, suite.chainA.Codec, suite.store, clientMsg)
-				suite.Require().NoError(err)
+			"misbehaviour found", func() {
+				suite.mockVM.RegisterQueryCallback(types.CheckForMisbehaviourMsg{}, func(codeID wasmvm.Checksum, env wasmvmtypes.Env, queryMsg []byte, store wasmvm.KVStore, goapi wasmvm.GoAPI, querier wasmvm.Querier, gasMeter wasmvm.GasMeter, gasLimit uint64, deserCost wasmvmtypes.UFraction) ([]byte, uint64, error) {
+					resp, err := json.Marshal(types.CheckForMisbehaviourResult{FoundMisbehaviour: true})
+					suite.Assert().NoError(err)
```

Comment:
```suggestion
					suite.Require().NoError(err)
```
I don't really know the difference but I think we always use `Require`